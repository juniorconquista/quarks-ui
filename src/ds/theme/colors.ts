export default {
    dark: false,
    primary: {
        default: '#F26522',
        dark: '#d14b0c',
        light: '#f69263',
        contrast: '#ffffff',
    },
    secondary: {
        default: '#372E2A',
        dark: '#2c2522',
        light: '#453935',
        contrast: '#1B2B3A',
    },
    danger: {
        default: '#ff6961',
        dark: '#ff261b',
        light: '#ffbdb9',
        contrast: '#ffffff',
    },
    success: {
        default: '#0E9d51',
        dark: '#0a753c',
        light: '#10bc60',
        contrast: '#ffffff',
    },
    warning: {
        default: '#ffb143',
        dark: '#e4902c',
        light: '#ffd868',
        contrast: '#ffffff',
    },
    info: {
        default: '#0ea7a7',
        dark: '#0a7676',
        light: '#11d4d4',
        contrast: '#ffffff',
    },
    grays: {
        default: '#9ba8b2',
        dark: '#4b4b4b',
        light: '#eeeeef',
        contrast: '#f5f5f5',
    },
    background: {
        default: '#ffffff',
        dark: '#F26522',
    },
    text: {
        light: '#ffffff',
        dark: '#1B2b3a',
    },
    disabled: '#9398a6',
    primaries: ['#d7edf4'],
    whites: ['#eeeeef', '#f2fbfe'],
    blacks: ['#35363a'],
    hovers: ['#f5f7f7'],
};
