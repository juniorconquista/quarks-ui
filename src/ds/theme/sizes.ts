export const breakpoints = ['0rem', '37.5rem', '62rem'];

export const space = {
    none: '0',
    xxxs: '0.285rem',
    xxs: '0.571rem',
    xs: '0.857rem',
    sm: '1.142rem',
    md: '1.714rem',
    lg: '2.5rem',
    xl: '3.428rem',
    xxl: '6.857rem',
};

export const fontSizes = [
    '0.5rem',
    '0.85rem',
    '1rem',
    '1.5rem',
    '1.8rem',
    '2rem',
];

export const buttonSizes = {
    medium: {
        paddingTop: 8,
        paddingBottom: 8,
        paddingRight: 16,
        paddingLeft: 16,
        borderRadius: 16,
    },
    small: {
        paddingTop: 4,
        paddingBottom: 4,
        paddingRight: 16,
        paddingLeft: 16,
        borderRadius: 16,
    },
};

export const radii = {
    none: '0',
    sm: '0.285rem',
    md: '2.25rem',
    lg: '1.142rem',
    xl: '1.714rem',
};
