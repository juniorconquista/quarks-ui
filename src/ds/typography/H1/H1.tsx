import styled from 'styled-components';
import { variant, VariantArgs, color, typography, space } from 'styled-system';
import { H1StyledProps } from './h1.types';

const variants = variant({
    prop: 'styling',
    variants: {
        base: {
            fontFamily: 'OpenSans Semibold',
            color: 'secondary.default',
            textAlign: 'left',
            fontSize: '40px',
            lineHeight: '54px',
        },
        get light() {
            return {
                ...this.base,
                fontFamily: 'OpenSans Light',
            };
        },
        get error() {
            return {
                ...this.base,
                color: 'danger.primary',
            };
        },
        get baseLight() {
            return {
                ...this.base,
                fontFamily: 'OpenSans Regular',
            };
        },
    },
} as VariantArgs);

export const H1: H1StyledProps = styled.h1`
    ${variants}
    ${color}
    ${typography}
    ${space}
`;

export default H1;
