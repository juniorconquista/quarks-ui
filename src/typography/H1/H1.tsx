import { SFC } from 'react';
import styled from 'styled-components';
import { variant, VariantArgs, color, typography, space } from 'styled-system';
import { H1Props } from './h1.types';

const variants = variant({
    prop: 'styling',
    variants: {
        base: {
            color: 'text.default',
            fontSize: 8,
        },
        get light() {
            return {
                ...this.base,
                fontWeight: 'normal',
            };
        },
    },
} as VariantArgs);

export const H1: SFC<H1Props> = styled.h1`
    ${variants}
    ${color}
    ${typography}
    ${space}
`;

export default H1;
