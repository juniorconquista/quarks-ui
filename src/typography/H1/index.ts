export { default } from './H1';
export * from './h1.types';
