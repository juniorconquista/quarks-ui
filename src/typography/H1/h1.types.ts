import { DefaultStyleProps } from '../../theme/types'
import { TypographyProps } from 'styled-system'

export type H1StyledProps = DefaultStyleProps & TypographyProps

export interface H1Props extends H1StyledProps {
  /** Each styling variant has unique properties that are inherited through the type informed */
  styling?: 'base' | 'light'
}
