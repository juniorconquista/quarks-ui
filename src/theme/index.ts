import colors from './colors';
import { breakpoints, space, fontSizes, radii } from './sizes';

const theme = {
    fontSizes,
    space,
    breakpoints,
    radii,
    colors: colors(),
};

//@ts-ignore
theme.breakpoints.mobile = breakpoints[0];
//@ts-ignore
theme.breakpoints.tablet = breakpoints[1];
//@ts-ignore
theme.breakpoints.desktop = breakpoints[2];

export default theme;
