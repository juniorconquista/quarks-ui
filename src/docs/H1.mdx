---
name: H1
menu: Components
---

import { ThemeProvider } from 'styled-components';
import { Playground, Props } from 'docz';
import { H1 } from '../ds/typography';
import * as theme from '../ds/theme';

# H1

## Properties

<Props of={H1} />

## Basic usage

<Playground>
    <ThemeProvider theme={{ ...theme.default }}>
        <H1 styling="base">Base</H1>
        <H1 styling="light">Base</H1>
    </ThemeProvider>
</Playground>
