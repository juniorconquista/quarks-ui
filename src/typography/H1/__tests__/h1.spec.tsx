import React from 'react';
import renderer from 'react-test-renderer';
import ThemeTest from '../../../utils/ThemeProviderTest';
import 'jest-styled-components';

import H1 from '../H1';
import theme from '../../../theme';

describe('H1', () => {
    test('basic snapshot', () => {
        const h1 = renderer
            .create(
                <ThemeTest>
                    <H1 styling="base" />
                </ThemeTest>,
            )
            .toJSON();
        expect(h1).toMatchSnapshot();
    });

    test('should inherit the styling properties of the base variant', () => {
        const h1 = renderer
            .create(
                <ThemeTest>
                    <H1 styling="base" />
                </ThemeTest>,
            )
            .toJSON();
        expect(h1).toHaveStyleRule('color', theme.colors.text.default);
        expect(h1).toHaveStyleRule('font-size', theme.fontSizes[8]);
    });

    test('should inherit the styling properties of the light variant', () => {
        const h1 = renderer
            .create(
                <ThemeTest>
                    <H1 styling="light" />
                </ThemeTest>,
            )
            .toJSON();
        expect(h1).toHaveStyleRule('color', theme.colors.text.default);
        expect(h1).toHaveStyleRule('font-size', theme.fontSizes[8]);
        expect(h1).toHaveStyleRule('font-weight', 'normal');
    });

    test('should apply custom styling properties', () => {
        const h1 = renderer
            .create(<H1 styling="base" mt="20px" p="3px" />)
            .toJSON();
        expect(h1).toHaveStyleRule('margin-top', '20px');
        expect(h1).toHaveStyleRule('padding', '3px');
    });
});
