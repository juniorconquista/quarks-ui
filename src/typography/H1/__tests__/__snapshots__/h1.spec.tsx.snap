// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H1 basic snapshot 1`] = `
.c0 {
  color: #343a40;
  font-size: 2rem;
}

<h1
  className="c0"
/>
`;
